phpdotnet/phd/Package/Generic/XHTML.php

```diff
@@ -2538,4 +2542,22 @@ public function format_whitespace($whitespace, $elementStack, $currentDepth) {
     public function format_caption($open, $name, $attrs, $props) {
         return $open ? '<div class="caption">' : '</div>';
     }
+
+    public function getGeneratedExampleID($index)
+    {
+        $originalId = parent::getGeneratedExampleID($index);
+        if (! $this->exampleCounterIsPerPage) {
+            return $originalId;
+        }
+        if (preg_match('/^example\-[0-9]+$/', $originalId)) {
+            $this->perPageExampleCounter++;
+            return 'example-' . $this->perPageExampleCounter;
+        }
+        return $originalId;
+    }
+
+    public function onNewPage(): void
```

Comment:
`onNewPage()` should test `exampleCounterIsPerPage`, otherwise it will always reset, even with `exampleCounterIsPerPage` is false.